crates/red_knot_python_semantic/src/types.rs

```diff
@@ -471,6 +471,232 @@ impl<'db> Type<'db> {
+                if intersection
+                    .positive(db)
+                    .iter()
+                    .any(|p| p.is_disjoint_from(db, other))
+                {
+                    true
+                } else {
+                    // TODO we can do better here. For example:
+                    // X & ~Literal[1] is disjoint from Literal[1]
+                    false
+                }
+            }
+
+            (Type::Tuple(tuple), other) | (other, Type::Tuple(tuple)) => {
+                if let Type::Tuple(other_tuple) = other {
+                    if tuple.len(db) == other_tuple.len(db) {
+                        tuple
+                            .elements(db)
+                            .iter()
+                            .zip(other_tuple.elements(db).iter())
+                            .any(|(e1, e2)| e1.is_disjoint_from(db, *e2))
+                    } else {
+                        true
+                    }
+                } else {
+                    true
+                }
+            }
+
+            (Type::None, other) | (other, Type::None) => match other {
+                Type::Never
+                | Type::Any
+                | Type::Unknown
+                | Type::Unbound
+                | Type::Todo
+                | Type::Function(..)
+                | Type::Module(..)
+                | Type::Class(..)
+                | Type::Union(..)
+                | Type::Intersection(..)
+                | Type::Tuple(..) => {
+                    unreachable!("handled above")
+                }
+                Type::IntLiteral(_)
+                | Type::BooleanLiteral(_)
+                | Type::StringLiteral(..)
+                | Type::LiteralString
+                | Type::BytesLiteral(..) => true,
+                Type::None => false,
+                Type::Instance(class_type) => !class_type.is_known(db, KnownClass::NoneType),
+            },
+
+            (Type::BooleanLiteral(bool), other) | (other, Type::BooleanLiteral(bool)) => {
+                match other {
+                    Type::Never
+                    | Type::Any
+                    | Type::Unknown
+                    | Type::Unbound
+                    | Type::Todo
+                    | Type::Function(..)
+                    | Type::Module(..)
+                    | Type::Class(..)
+                    | Type::Union(..)
+                    | Type::Intersection(..)
+                    | Type::Tuple(..)
+                    | Type::None => {
+                        unreachable!("handled above")
+                    }
+                    Type::IntLiteral(_)
+                    | Type::StringLiteral(..)
+                    | Type::LiteralString
+                    | Type::BytesLiteral(..) => true,
+                    Type::BooleanLiteral(bool_other) => bool != bool_other,
+                    Type::Instance(class_type) => !class_type.is_known(db, KnownClass::Bool),
```

Comment:
Same question here.

Reply:
Same issue here. `Literal[True]` is a subtype of `bool`, but `bool` is a subtype of `object` (all classes in Python are subclasses of `object`, either explicitly or implicitly)